// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: list_domains.proto

package identityd2

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"github.com/golang/protobuf/ptypes"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = ptypes.DynamicAny{}
)

// define the regex for a UUID once up-front
var _list_domains_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")

// Validate checks the field values on ListDomainsRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListDomainsRequest) Validate() error {
	if m == nil {
		return nil
	}

	if v, ok := interface{}(m.GetDomain()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ListDomainsRequestValidationError{
				field:  "Domain",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// ListDomainsRequestValidationError is the validation error returned by
// ListDomainsRequest.Validate if the designated constraints aren't met.
type ListDomainsRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListDomainsRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListDomainsRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListDomainsRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListDomainsRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListDomainsRequestValidationError) ErrorName() string {
	return "ListDomainsRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ListDomainsRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListDomainsRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListDomainsRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListDomainsRequestValidationError{}

// Validate checks the field values on ListDomainsResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListDomainsResponse) Validate() error {
	if m == nil {
		return nil
	}

	for idx, item := range m.GetDomains() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ListDomainsResponseValidationError{
					field:  fmt.Sprintf("Domains[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// ListDomainsResponseValidationError is the validation error returned by
// ListDomainsResponse.Validate if the designated constraints aren't met.
type ListDomainsResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListDomainsResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListDomainsResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListDomainsResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListDomainsResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListDomainsResponseValidationError) ErrorName() string {
	return "ListDomainsResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ListDomainsResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListDomainsResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListDomainsResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListDomainsResponseValidationError{}
